akka-projection-cassandra/src/main/scala/akka/projection/cassandra/internal/CassandraProjectionImpl.scala

```diff
@@ -81,28 +42,26 @@ import akka.stream.scaladsl.Source
     sourceProvider: SourceProvider[Offset, Envelope],
     settingsOpt: Option[ProjectionSettings],
     restartBackoffOpt: Option[RestartBackoffSettings],
-    val offsetStrategy: CassandraProjectionImpl.OffsetStrategy,
-    handlerStrategy: CassandraProjectionImpl.HandlerStrategy[Envelope],
+    val offsetStrategy: OffsetStrategy,
+    handlerStrategy: HandlerStrategy[Envelope],
     override val statusObserver: StatusObserver[Envelope])
-    extends javadsl.CassandraProjection[Envelope]
-    with scaladsl.CassandraProjection[Envelope]
-    with javadsl.AtLeastOnceCassandraProjection[Envelope]
-    with scaladsl.AtLeastOnceCassandraProjection[Envelope]
-    with javadsl.GroupedCassandraProjection[Envelope]
-    with scaladsl.GroupedCassandraProjection[Envelope]
-    with javadsl.AtMostOnceCassandraProjection[Envelope]
-    with scaladsl.AtMostOnceCassandraProjection[Envelope]
-    with scaladsl.AtLeastOnceFlowCassandraProjection[Envelope]
-    with javadsl.AtLeastOnceFlowCassandraProjection[Envelope]
+    extends scaladsl.CassandraProjection[Offset, Envelope]
+    with javadsl.CassandraProjection[Offset, Envelope]
+    with scaladsl.AtLeastOnceCassandraProjection[Offset, Envelope]
```

Comment:
do we still need `CassandraProjection` trait and `AtLeastOnceCassandraProjection`? Would it be enough with `AtLeastOnceProjection` from core?

Reply:
yes, that's because of the createTable method. That's the reason all that duplication. 

We need to keep it a `CassandraProjection` (or Slick) all the way, but without removing the specific at-least, at-most, etc.

Reply:
We could move the create table to a utility class, than this would be solved.

I can do it on another PR.

Reply:
> We could move the create table to a utility class, than this would be solved.

I had that on my TODO list anyway, because I found that it's not convenient to have it on the Projection when used from tests where you typically use it once before running the tests. Moving to utility would be good.

In this PR you could just remove the `createOffsetTableIfNotExists`. It's not used anywhere. Then we add it in utility in separate PR.

Reply:
Decided to not send yet another PR. It's integrated here: 4f8e55e28bf8bfb8b44f0a6675e1a8e5cfb9e5a2